### DISCUSSION TITLE
Two Web Cameras Video Capture

### DISCUSSION THREAD

**Discussion by sergiibulatov** [emgucv/emgucv#423] [Q&A]
Hi,

I am experiencing strange issue. I have two web cameras (video microscope) and would like to display video simultaneously from both of them. On my PC everything works as expected. When I run software with the same cameras on second PC only one camera can initialize, but never two. Second camera always has IsOpened false. Debuging on the second PC doesn't show the root cause. Than I run software on two others PCs (laptop and desktop), laptop worked desktop not.
I use Emgu.CV 4.4.0.4099 (NuGet package), WPF application, Visual Studio 2019 (Community), all PCs have latest Windows 10 update. Tried with API: DShow, Any (with Any initialization takes almost one minute and doesn't open second camera). 
Could you please suggest direction for troubleshooting?
Thank you.

Sergii

**Comment by emgucv** [comment]
Probably issue with DShow. You can verify it with DirectShowNet: http://directshownet.sourceforge.net/ 

Does the camera required driver installation for DShow?

If it is a problem with Direct Show, it will need to be fixed on Direct Show side. 

Can the second camera (that failed to open on the laptop) be opened on the PC using other camera apps (that use DShow interface)?

**Reply by sergiibulatov** [reply]
Windows automatically detects camera and set camera driver (10.0.19041.488). No driver request for DShow.
It is possible to capture with AmCap and Windows 10 Camera App for that camera.
VideoCapture(0) works when it is the only camera in the system, when two or more cameras, camera stays unopened. I checked camera indexes up to 1000 in case some enumeration issue exist, without success. I found that even on my PC happens that sometimes software doesn't open camera.
